packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -842,7 +783,6 @@ export function diffHydratedProperties(
   let extraAttributeNames: Set<string>;
 
   if (__DEV__) {
-    suppressHydrationWarning = rawProps[SUPPRESS_HYDRATION_WARNING] === true;
```

Comment:
The `suppressHydrationWarning` variable was file-level for some reason, not function-level, so it may have affected functions unrelated to this function call. Now its expression `rawProps[SUPPRESS_HYDRATION_WARNING] === true` is only used in one place within this function, so the variable was removed.